### PR TITLE
Add project 4 with GraphQL

### PR DESCRIPTION
**Pull request by pirvudoru**

### CODE REVIEW

File: projects/04-training-api/requirements.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Twitter API
+
+The purpose of the this project is to expose a GraphQL API.
+
+# Resources
+
+Go through the following before starting the implementation
+
+- [GraphQL Crash Course](https://www.youtube.com/watch?v=YFS5fAH2zFQ)
```
Comment (alex-damian-negru) [inline projects/04-training-api/requirements.md:9]: Also recommend [this](https://www.youtube.com/watch?v=eIQh02xuVw4) & everything from that channel.

File: projects/04-training-api/examples/nokogiri_example.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require 'nokogiri'
+# XPath: https://devhints.io/xpath
+# Nokogiri: https://nokogiri.org/#parsing-and-querying
+
+require 'open-uri' # Load content from URL
+
+doc = Nokogiri::HTML(open('https://12ft.io/'))
+doc.xpath('//meta[@property="og:url"]').first['content']
```
Comment (alex-damian-negru) [inline projects/04-training-api/examples/nokogiri_example.rb]: Using Nokogiri implies knowledge of XML. If we go for this, note that we might also have to explain XML.

It shouldn't be something hard, but we'll introduce one more concept (or two if we count Nokogiri), and I think GraphQL has quite a few already.

That aside, they might get confused by Nokogiri syntax.

**Review by alex-damian-negru — COMMENTED** [review]
An alternative, more easier idea might be the classic _blog_. The tutorial also has [examples](https://graphql-ruby.org/getting_started.html) for this.

**Review by drqCode — APPROVED** [review]

File: projects/04-training-api/examples/nokogiri_example.rb (hunk 2 above)
Comment (pirvudoru) [reply projects/04-training-api/examples/nokogiri_example.rb]: Improved the example. We can answer questions around it, but that part should not be a blocker IMO

**Pull request merged**